jaeger_client/span.py

```diff
@@ -32,23 +37,30 @@ class Span(opentracing.Span):
                  'operation_name', 'start_time', 'end_time',
                  'logs', 'tags', 'finished', 'update_lock']
 
-    def __init__(self, context, tracer, operation_name,
-                 tags=None, start_time=None, references=None):
+    def __init__(
+        self,
+        context: SpanContext,
+        tracer: 'Tracer',
+        operation_name: str,
+        tags: Optional[Dict[str, Any]] = None,
+        start_time: Optional[float] = None,
+        references: Optional[Any] = None
+    ) -> None:
         super(Span, self).__init__(context=context, tracer=tracer)
         self.operation_name = operation_name
         self.start_time = start_time or time.time()
-        self.end_time = None
+        self.end_time: Optional[float] = None
         self.finished = False
         self.update_lock = threading.Lock()
         self.references = references
         # we store tags and logs as Thrift objects to avoid extra allocations
-        self.tags = []
-        self.logs = []
+        self.tags: list = []
```

Comment:
list of thrift types?

Reply:
So the tags are a list of thrift types?

Reply:
they are created via `thrift.make_tag()`

Reply:
So the actual type should be?

Reply:
```suggestion
        self.tags: List[ttypes.Tag] = []
```